source/type/pb_converter.h

```diff
@@ -34,6 +34,85 @@ www.navitia.io
 #include "type/response.pb.h"
 #include "type/pt_data.h"
 
+namespace pt = boost::posix_time;
+
+// we don't use a boolean to have a more type checks
+enum class DumpMessage {
+    Yes,
+    No
+};
+
+#define null_time_period boost::posix_time::time_period(boost::posix_time::not_a_date_time, boost::posix_time::seconds(0))
+
+//*********************************************
+
+using house_number_coord = std::pair<int, navitia::type::GeographicalCoord>;
+using way_name = std::pair<const navitia::georef::Way*, const std::string>;
+using way_pair = std::pair<const navitia::georef::Way*, house_number_coord>;
+using way_pair_name = std::pair<way_name, house_number_coord>;
+
+namespace ProtoCreator {
+struct PbCreator {
+    const nt::Data& data;
+    const pt::ptime now;
+    const pt::time_period action_period;
+    const bool show_codes;
+    PbCreator(const nt::Data& data, const pt::ptime  now, const pt::time_period action_period, const bool show_codes):
+        data(data), now(now), action_period(action_period),show_codes(show_codes) {}
+    template<typename N, typename P>
+    void fill(int depth, const DumpMessage dump_message, const N* item, P * proto) {
+        Filler(depth, dump_message, *this).filler_pb_object(item, proto);
+    }
+
+private:
+    struct Filler {
+        int depth;
+        const DumpMessage dump_message;
+        PbCreator & pb_creator;
+        Filler(int depth, const DumpMessage dump_message, PbCreator & pb_creator):
+            depth(depth), dump_message(dump_message), pb_creator(pb_creator){};
+
+        Filler copy_depth_decr();
+        template<typename NAV, typename PB>
+        void fill(const NAV* nav_object, PB* pb_object) {
+            copy_depth_decr().filler_pb_object(nav_object, pb_object);
+        }
```

Comment:
A little suggestion to make it more generic:

       
       template<typename ...Args>
       void fill(Args&&... args) {
           copy_depth_decr().filler_pb_object(std::forward<Args>(args)...);
       }